doc/concepts/moveit_task_constructor/moveit_task_constructor.rst

```diff
@@ -0,0 +1,212 @@
+| The most important generator stage is `CurrentState`, which provides the current robot state as the starting point for a planning pipeline.
+
+| Monitoring Generator - Generator that monitors the solution of another stage to make reuse of them.
+| Main example of Monitoring Generator - `GeneratePose`. It usually monitors a `CurrentState` or `ModifyPlanning Scene` stage.
+
+Currently available generator stages:
+
+* CurrentState
+
+* FixedState
+
+* Monitoring Generators - GeneratePose, GenerateGraspPose, GeneratePlacePose, GenerateRandomPose
+
+| Link for more information on generator stages available to use :ref:`Generating Stages`.
+
+Propagating Stage
+^^^^^^^^^^^^^^^^^
+| Propagators receive the result from one neighbor state, solve a subproblem and then propagate the result to the neighbor on the opposite side.
+| Depending on the implementation, this stage can pass solutions forward, backward or in both directions.
+
+| Link for more information on propagating stages available :ref:`Propagating Stages`.
+
+Connecting Stage
+^^^^^^^^^^^^^^^^
+Connectors do not propagate any results but attempt to bridge the gap between the resulting states of both neighbors.
+
+| Link for more information on how to use connect stage :ref:`Connecting Stages`.
+
+Wrapper
+^^^^^^^
+| Wrappers encapsulate a single stage and modify/filter the results.
+| Currently available Wrappers - ComputeIK, PredicateFilter and PassThrough
+
+| Link for more information on wrappers available to use :ref:`Wrappers`.
+
+MTC Containers
+---------------
+The MTC framework enables the hierarchical organization of stages using containers, allowing for sequential as well as parallel compositions.
+
+Serial Container
+^^^^^^^^^^^^^^^^
+Serial containers hold a sequence of stages (and only consider end-to-end solutions as results)
+A MTC Task by default is stored as a serial container.
+
+| Link for more information on how to use serial container : TBD
+
+Parallel Container
+^^^^^^^^^^^^^^^^^^
+Parallel containers combine a set of stages in the following formats
+
+* Alternatives - Solution of all children are collected and sorted by cost.
+
+* Fallback - A fallback container executes children stages in order until one of them returns success or all stages return failure
+
+* Merger - Solutions of all children (actuating disjoint groups) are planned and executed parallelly.
+
+| Link for more information on how to use parallel container : TBD
+
+Initializing a MTC Task
+-----------------------
+
+The top-level planning problem is specified as a MTC Task and the subproblems which are specified by Stages are added to the MTC task object.
+
+.. code-block:: c++
+
+  auto node = std::make_shared<rclcpp::Node>();
+  auto task = std::make_unique<moveit::task_constructor::Task>();
+  task->loadRobotModel(node);
+  // Set controllers used to execute robot motion
+  task->setProperty("trajectory_execution_info", "joint_trajectory_controller gripper_controller");
```

Comment:
Should we say this is optional and that MoveIt does its own discovery otherwise? And generally, is this worth a small subsection describing this behavior?